pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+        return self._schema_ver
+
+    @property
+    def schema_ver_str(self):
+        """
+        :term:`string` with the DMTF schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_ver[0], self.schema_ver[1],
+                             self.schema_ver[2])
+
+    @property
+    def schema_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is downloaded and expanded.
+        """
+        return self._schema_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is defined. This includes the expansion of the individual schema
+        mof files. This property is very useful since it can be used as
+        the search path for compiling classes in the DMTF schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string` defining the the path for the DMTF schema
+        mof top level file name which includes the pragmas that define
+        all of the files in the DMTF schema. This filename is of the
+        general form
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFSchema` object for human consumption.
+        """
+
+        return '%s(schema_ver=%s, schema_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_ver_str,
+                self.schema_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the :class:`~pywbem_mock.DMTFSchema`
+        object that is suitable for debugging.
+
+        The properties and qualifiers will be ordered by their names in the
+        result.
+        """
+
+        return '{0}(schema_ver={1}, schema_dir={2})'.format(
+            self.__class__.__name__, self.schema_ver, self.schema_dir)
+
+    def _setup_dmtf_schema(self):
+        """
+        Install the DMTF schema from the DMTF web site if it is not
+        already installed.
+
+        Once the schema in installed in 'schema_dir', it is no longer touched if
+        this function is recalled.  Further, if the zip file is in the schema
+        directory, but there no mof subdirectory, has been created
+        the schema is unzipped without downloading the zip file from the
+        DMTF.
+
+        This allows the dmtf schema to be downloaded once and reused and further
+        allows a much smaller footprint for the schema for repositories, etc by
+        unzipping the schema zip file if it has not already been unzipped.
+
+        Raises:
+            IOError if the remote url cannot be opened.
+        """
+        first = [True]  # List allows setting variable from inner function
+
+        def if_first(msg):
```

Comment:
The name of this inner function should indicate that it prints. And it always prints, not just if first.

Reply:
print_verbose it is. DONE